sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -158,6 +159,28 @@
  * Read#withValueDeserializerAndCoder(Class, Coder)}. Note that Kafka messages are interpreted using
  * key and value <i>deserializers</i>.
  *
+ * <h3>Read From Kafka Dynamically</h3>
+ *
+ * For a given kafka bootstrap_server, KafkaIO is also able to detect and read from available {@link
+ * TopicPartition} dynamically. For more design details, please refer to
+ * https://docs.google.com/document/d/1FU3GxVRetHPLVizP3Mdv6mP5tpjZ3fd99qNjUI5DT5k/. To enable
```

Comment:
Please, put the details here (at least in a short form) instead of the link since it can be expired or disappeared along the time.

Reply:
Will do. Thanks for pointing this out.